lib/ansible/module_utils/basic.py

```diff
@@ -1426,7 +1419,7 @@ def _handle_aliases(self, spec=None, param=None, option_prefix=''):
         alias_warnings = []
         alias_results, self._legal_inputs = handle_aliases(spec, param, alias_warnings=alias_warnings)
         for option, alias in alias_warnings:
-            self._warnings.append('Both option %s and its alias %s are set.' % (option_prefix + option, option_prefix + alias))
+            warnings.global_warnings.append('Both option %s and its alias %s are set.' % (option_prefix + option, option_prefix + alias))
```

Comment:
Why are you not using the helper functions here?

Reply:
I think I missed this one. I had a deep discussion with @abadger about the nuances of importing globals. I'm pretty sure I need to use the helper function and not modify the var directly.